met-api/src/met_api/models/engagement.py

```diff
@@ -233,11 +233,29 @@ def _filter_by_search_text(query, search_options):
         return query
 
     @staticmethod
-    def _filter_by_project_type(query, project_type=None):
-        if project_type:
-            query = query.outerjoin(EngagementMetadataModel, EngagementMetadataModel.engagement_id == Engagement.id)\
-                .filter(EngagementMetadataModel.project_metadata['type'].astext == project_type)\
+    def _filter_by_project_metadata(query, search_options):
+        query = query.outerjoin(EngagementMetadataModel, EngagementMetadataModel.engagement_id == Engagement.id)
+
+        if project_type := search_options.get('project_type'):
+            query = query.filter(EngagementMetadataModel.project_metadata['type'].astext == project_type)\
                 .params(val=project_type)
+
+        if project_name := search_options.get('project_name'):
+            query = query.filter(EngagementMetadataModel.project_metadata['project_name'].astext == project_name)\
```

Comment:
All the text fields should be a partial match e.g. project_name like '%test%'

Reply:
Good point ill do that then